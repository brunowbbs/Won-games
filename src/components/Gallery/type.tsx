export type GaleryImageProps = {
  src: string;
  label: string;
};

export type GalleryProps = {
  items: GaleryImageProps[];
};
