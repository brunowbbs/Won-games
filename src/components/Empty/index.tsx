import Link from "next/link";
import Button from "../Button";
import * as S from "./styles";
import { EmptyProps } from "./type";

const Empty = ({ title, description, hasLink }: EmptyProps) => {
  return (
    <S.Wrapper>
      <S.Image
        src="img/empty.svg"
        role="img"
        alt="A gammer in a couch playing videogame"
      />

      <S.Title>{title}</S.Title>

      <S.Description>{description}</S.Description>

      {hasLink && (
        <Link href="/" passHref>
          <Button as="a">Go back to store</Button>
        </Link>
      )}
    </S.Wrapper>
  );
};

export default Empty;
