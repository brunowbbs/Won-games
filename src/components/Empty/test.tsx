import { screen } from "@testing-library/react";
import { renderWithTheme } from "../../utils/tests/helpers";
import Empty from ".";
import { EmptyProps } from "./type";

const props: EmptyProps = {
  title: "A simple title",
  description: "A simple description",
};

describe("<Empty/>", () => {
  it("Should render correctly", () => {
    const { container } = renderWithTheme(<Empty {...props} hasLink />);
    expect(
      screen.getByRole("img", {
        name: /a gammer in a couch playing videogame/i,
      })
    ).toBeInTheDocument();

    expect(
      screen.getByRole("heading", { name: /a simple title/i })
    ).toBeInTheDocument();

    expect(screen.getByText(/a simple description/i)).toBeInTheDocument();

    expect(
      screen.getByRole("link", { name: /go back to store/i })
    ).toHaveAttribute("href", "/");

    expect(container.firstChild).toMatchSnapshot();
  });

  it("Should not render link when haslink is not passed", () => {
    renderWithTheme(<Empty {...props} />);

    expect(
      screen.queryByRole("link", { name: /go back to store/i })
    ).not.toBeInTheDocument();
  });
});
