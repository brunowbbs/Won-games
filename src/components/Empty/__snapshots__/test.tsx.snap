// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Empty/> Should render correctly 1`] = `
<div
  class="sc-gswNZR icgcca"
>
  <img
    alt="A gammer in a couch playing videogame"
    class="sc-dkrFOg jwHEtL"
    role="img"
    src="img/empty.svg"
  />
  <h2
    class="sc-hLBbgP lcWAtF"
  >
    A simple title
  </h2>
  <p
    class="sc-eDvSVe dAcHZr"
  >
    A simple description
  </p>
  <a
    class="sc-bcXHqe hnaJZy"
    href="/"
  >
    <span>
      Go back to store
    </span>
  </a>
</div>
`;
